website/views.py

```diff
@@ -214,10 +215,10 @@ def resolve_guid(guid, suffix=None):
         if not referent.deep_url:
             raise HTTPError(http.NOT_FOUND)
         if isinstance(referent, PreprintService):
-            return send_from_directory(
-                os.path.abspath(os.path.join(os.getcwd(), EXTERNAL_EMBER_APPS['preprints']['path'])),
-                'index.html'
-            )
+            if referent.provider._id == 'osf' or not referent.provider.domain:
```

Comment:
That's a fair question! 

In truth, GUIDs (and related tests) are wound through many parts of the OSF, which makes it hard to find a single obvious place to put the test.

Here are some general GUID tests:
https://github.com/abought/osf.io/blob/5ad6ebe27e63649213801cee9c395f665fb84056/tests/test_guids.py#L36-L36

Or sometimes GUID-related functionality will get tested with the module for that specific feature. Here are a few examples; there are 279 total references in the `tests` folder, but it's not at all centralized.
Addons: https://github.com/abought/osf.io/blob/33eff35d7fadc322dc4fcd1f9c885e48366a9681/tests/test_addons.py#L656-L656

Comments: https://github.com/abought/osf.io/blob/41484ecfd66a69ebaeeeb5db8183867c28c896d1/tests/test_comments.py#L137-L137

Models: https://github.com/abought/osf.io/blob/e17666d666beb60dbabfb4eaf652efe523300460/tests/test_models.py#L1051-L1051